File: src/binarytree/lc_814_binary_tree_pruning/binary_tree_pruning.go
type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}
	
func pruneTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	if root.Val == 0 && root.Left == nil && root.Right == nil {
		return nil
	}
	node := new(TreeNode)
	node.Val = root.Val
	node.Left = pruneTree(root.Left, target)
	node.Right = pruneTree(root.Right, target)
	if node.Left == nil && node.Right == nil && node.Val == 0 {
		return nil
	}
	return node
}

File: src/binarytree/lc_814_binary_tree_pruning/binary_tree_pruning.py
from treeNode import TreeNode
class Solution:
    def pruneTree(self, root: TreeNode) -> TreeNode:
        def dfs(node):
            if not node:
                return None

            if node.val == 0 and not node.left and not node.right:
                return None
            
            root = TreeNode(node.val)
            root.left = dfs(node.left)
            root.right = dfs(node.right)
            if not root.right and not root.left and root.val == 0:
                return None
            return root
        return dfs(root)